backend-diagram-converter/core/src/main/java/org/camunda/community/migration/converter/visitor/impl/element/InputOutputParameterVisitor.java

```diff
@@ -27,14 +27,25 @@ protected Message visitCamundaElement(DomElementVisitorContext context) {
     if (isNotStringOrExpression(element)) {
       return MessageFactory.inputOutputParameterIsNoExpression(localName(), name);
     }
+    String expression = element.getTextContent();
     ExpressionTransformationResult transformationResult =
-        ExpressionTransformer.transform(element.getTextContent());
+        ExpressionTransformer.transform(expression);
     context.addConversion(
         AbstractDataMapperConvertible.class,
         abstractTaskConversion ->
             abstractTaskConversion.addZeebeIoMapping(
                 direction, transformationResult.getNewExpression(), name));
-    return MessageFactory.inputOutputParameter(localName(), name, transformationResult);
+    Message resultMessage;
+    if (transformationResult.hasExecution()) {
+      resultMessage =
+          MessageFactory.inputOutputParameterExecution(localName(), name, transformationResult);
+    } else if (transformationResult.hasMethodInvocation()) {
```

Comment:
This should be an if-logic without else as the cases do not exclude each other. This would lead to potentially multiple messages added per expression, but the result is deterministic.

Reply:
Same here.